CHANGELOG.md

```diff
@@ -1,5 +1,218 @@
+ERROR: too many arguments
+$ kontena stack rm example-stack --force
+ERROR: too many arguments
+```
+
+Note that if you need to use something that looks like an option as a parameter you need to use the
+common double dash `--` option break indicator:
+
+```
+$ kontena master ssh -- ls -al
+ERROR: Unrecognised option '-l'
+$ kontena master ssh -- ls -al
+$ kontena vault write -- SECRET --secret-password--
+```
+
+##### Kontena Stack Registry V2 API And The New 'meta' Fields
+
+While mostly invisible to the end-user, the CLI stack registry API client is now using
+the completely rewritten stack registry and the V2 JSON-API it offers. The registry
+supports GZip responses, private stacks, server-side stack YAML validation and parsing
+of the new top level 'meta:' fields.
+
+The meta fields can be used to add extra information to stacks published in the registry.
+
+You can find the full set of accepted metadata fields in the pull request #3219 description.
+
+As the CLI HTTP client now supports gzip compressed responses, we have also added the option
+to enable compression in the Kontena Master API. To enable, set `KONTENA_SERVER_GZIP=true`
+in the Master environment.
+
+##### Drop Support For Ruby 2.1, Build Installer With Embedded Ruby 2.5.0
+
+As Ruby 2.1 branch has been out of development for almost a year now, it's time to upgrade
+if you already didn't.
+
+The MacOS Kontena CLI installation package is now bundled with Ruby version 2.5.0
+
+Ruby 2.2 is nearing its EOL at the end of March 2018.
+
+##### Process Multiple Items In One Command
+
+Many of the subcommands can now accept a list of items instead of just one. This is handy in
+shell scripts and one-liners, for example:
+
+```
+$ kontena vault ls -q | xargs kontena vault rm --force
+$ kontena vault rm --force $(kontena vault ls -q)
+```
+
+### Changes
+
+#### Agent
+* Add health check port to LB configs (#3113)
+* Add Agent Watchdog supervisor to agent (#3135)
+* Fix agent ServicePodWorker to ignore stale container events (#3259)
+* Change agent health check to accept HTTP 3xx as healthy (#3265)
+* Log container healthcheck errors (#3284)
+* Log service:instance_exit event on container crashes (#3286)
+* Fix agent to unregister LB service backends earlier during container shutdown (#3287)
+* Fix agent container log dropping entries when queue size exactly matches the throttle limit (#3288)
+
+#### Agent + Server
+* Use GridService revision for service/container updates (#2371)
+* Improve agent ServicePodWorker container restart handling (#2780)
+
+#### Server
+* Remove server AsyncHelper#async_thread (#2786)
+* Fix service affinity filters to be stack-scoped (#2967)
+* Cap stack/service deploy collections (#3041)
+* Deploy tls-sni challenge certs as separate SSL_CERT_acme_challenge_* envs (#3076)
+* Support regex in affinity filters (#3099)
+* Remove dependant service logic (#3100)
+* Less shuffling when nodes missing with daemon strategy (#3124)
```

Comment:
Not sure if it makes sense to include closed/not-merged PRs in the changelog?

Reply:
No it does not.